internal/controller/streams.go

```diff
@@ -196,12 +193,38 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+
+	resourceVersions := make(map[string]string)
 	for resource, apiGroup := range resourceAPIGroupMap {
-		allResourcesSnapshotted.Add(1)
-		go resourceLister.DynamicListAndWatchResources(ctx, cancel, resource, apiGroup, &allResourcesSnapshotted, &snapshotCompleted)
+		// Add jitter between list resources
+		jitter := time.Duration(rand.Intn(1000)+1000) * time.Millisecond // 1-2 s
+		time.Sleep(jitter)
```

Comment:
1. This sleep is too long.
1. I think this sleep is unnecessary.
1. You shouldn't sleep while ignoring `ctx`. If we really needed to sleep here, this should be using a `Timer` and a select with another case to check `ctx.Done()`.

Remove this and check that it's alright:
```suggestion
```